### PR TITLE
Setup post resume review route

### LINKED ISSUE #75: Implement post resume review endpoint
`POST /resume-reviews`

### PR DESCRIPTION
**Pull request by calebschoepp**
# Overview

Setup post resume review route. Wanted to put up another example of implementing a route. The creation flow required some tweaking but this should be a more self-contained and readable PR

# Changes

- Implement post resume review route
- Make validator async so it can call to repository layer
- Add a user repository
- Write associated tests

# Questions & Notes

None as of yet

# Issue tracking

Closes #75

# Testing & Demo

- Wrote unit tests
- Turn off authentication middleware and use something like this `curl 'localhost:1337/api/v1/resume-reviews' -v -s --data '{"reviewee": "319ffe68-cac5-470f-9186-33371300c38f"}' -H 'Content-Type: application/json' | jq`

### CODE REVIEW

File: api/src/controllers/validation.ts [hunk 1]
```diff
@@ -43,11 +44,26 @@ const beAValidUuid = {
  * Test whether a field is of the union type resume_review_state.
  */
 const beAResumeReviewState = {
-    predicate: (field: string | undefined): boolean => {
+    predicate: async (field: string | undefined): Promise<boolean> => {
         return field ? ['canceled', 'finished', 'reviewing', 'seeking_reviewer'].includes(field) : false;
     },
     message: 'Must be a "canceled", "finished", "reviewing", or "seeking_reviewer"',
```
Comment (armiantos) [inline api/src/controllers/validation.ts:50]: Nit: change this to `Must be one of ...`?

File: api/src/controllers/validation.ts [hunk 2]
```diff
@@ -43,11 +44,26 @@ const beAValidUuid = {
...
+ * Test whether a user already exists in the database.
+ */
+const beAValidUser = {
+    predicate: async (field: string | undefined): Promise<boolean> => {
+        if (field === undefined) {
+            return false;
+        } else {
+            const matches = await userRepository.get(field);
+            return matches.length == 1;
+        }
```
Comment (armiantos) [inline api/src/controllers/validation.ts:63]: Nit: get rid of the else statement

**Review by armiantos — CHANGES_REQUESTED** [review]
Awesome job on this PR 👏

Just a couple nits from my end and one question regarding the test

File: api/src/controllers/resumeReview/postResumeReview.test.ts [hunk 3]
```diff
@@ -1 +1,75 @@
...
+        ccid: 'myccid',
+        program: 'compe',
+        year: 4,
+        given_name: 'bob',
+        family_name: 'saggit',
+        full_name: 'bob saggit',
+        photo_url: 'asdf.com',
+        created_at: '2021-06-14T06:09:19.373404+00:00',
+        updated_at: '2021-06-14T06:09:19.373404+00:00',
+    };
```
Comment (armiantos) [inline api/src/controllers/resumeReview/postResumeReview.test.ts:64]: These are just random values to get the repositories to return something valid right?

**Comment by cyrusdiego** [comment]
This LGTM, will let Armi approve though as he has some changes he requested. Good work on this, it's looking super clean :D

File: api/src/controllers/resumeReview/postResumeReview.test.ts (hunk 3 above)
Comment (calebschoepp) [reply api/src/controllers/resumeReview/postResumeReview.test.ts:64]: Yep random values for testing. In a future PR I might make a test constants file so we can just import sample users and reviews etc.

Comment (armiantos) [reply api/src/controllers/resumeReview/postResumeReview.test.ts:64]: Sounds good 👍

**Review by armiantos — APPROVED** [review]
LGTM!

**Pull request merged**